src/poetry/repositories/link_sources/html.py

```diff
@@ -44,7 +52,14 @@ def links(self) -> Iterator[Link]:
                 if link.ext not in self.SUPPORTED_FORMATS:
                     continue
 
-                yield link
+                pkg = self.link_package_data(link)
+                if pkg:
+                    links[pkg.name][pkg.version].append(link)
+                else:
+                    # dummy name and version
```

Comment:
This seems weird.  Is it useful?  Can this branch just be removed?

Reply:
I believe this branch was needed to be able to pass tests where the links provided in the backend were invalid, and thus package objects with no name/version were causing issues. I wasn't comfortable enough with exactly why that was being tested enough to remove them.

Reply:
The question is: "Do we want to return **all** links when calling `LinkSource.links` or only links representing valid package names? At the moment, we return all links. If we want to keep this behavior the branch is necessary.